### PR TITLE
Java: Revises `DEVELOPER.md` file.

### PR DESCRIPTION
**Pull request by tetrachromium**
Revises the `DEVELOPER.md` file, based on experience following the installation instructions, to reduce some potential for confusion and add additional troubleshooting information.

### CODE REVIEW

**Review by acarbonetto — COMMENTED** [review]
Please update merging branch to something like:
`java/integ_developer-doc-revisions`
which should be a branch off of `main`

File: java/DEVELOPER.md [hunk 1]
```diff
@@ -2,15 +2,17 @@
 
 This document describes how to set up your development environment to build and test the GLIDE for Redis Java wrapper.
 
-### Development Overview
+## Development Overview
```
Comment (acarbonetto) [inline java/DEVELOPER.md:5]: we want to preserve the styling between developer.md files: https://github.com/Bit-Quill/glide-for-redis/blob/main/python/DEVELOPER.md?plain=1#L5

**Review by acarbonetto — APPROVED** [review]
Please revert the style changes.  Otherwise, these changes look good.

**Review by Yury-Fridlyand — APPROVED** [review]
Thanks!

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**